File: adapter/ohos/HapVerifyInfo.java
/*
 * Copyright (c) 2022 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ohos;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * collection of HapVerifyInfo members,
 * those members will be verified in hapVerify.
 */
class HapVerifyInfo {
    /**
     * Indicates the bundleName of module.
     */
    private String bundleName = "";

    /**
     * Indicates the vendor of module.
     */
    private String vendor = "";

    /**
     * Indicates the version of module.
     */
    private Version version = new Version();

    /**
     * Indicates the apiVersion of module.
     */
    private ModuleApiVersion apiVersion = new ModuleApiVersion();

    /**
     * Indicates the assetAccessGroups of module.
     */
    private List<String> assetAccessGroups = new ArrayList<>();

    /**
     * Indicates the moduleName of module.
     */
    private String moduleName = "";

    /**
     * Indicates the package name of module.
     */
    private String packageName = "";

    /**
     * Indicates the ability names of module.
     */
    private List<String> abilityNames = new ArrayList<>();

    /**
     * Indicates the distrofilter of module.
     */
    private DistroFilter distroFilter = new DistroFilter();

    /**
     * Indicates the deviceType of module.
     */
    private List<String> deviceType = new ArrayList<>();

    /**
     * Indicates is stage of module.
     */
    private boolean isStageModule = true;

    /**
     * Indicates is type of module.
     */
    private String moduleType = "";

    /**
     * Indicates atomic service type, contain main, normal.
     */
    private String atomicServiceType = "";

    /**
     * Indicates is installationFree of module.
     */
    private boolean isInstallationFree = false;

    /**
     * Indicates dependency of module.
     */
    private List<String> dependencies = new ArrayList<>();

    /**
     * Indicates dependency of module.
     */
    private List<DependencyItem> dependencyItemList = new ArrayList<>();

    /**
     * Indicates is config.json string or module.json string of module.
     */
    private String profileStr = "";

    /**
     * Indicates is file in profile of module.
     */
    private HashMap<String, String> resourceMap = new HashMap<>();

    private String bundleType = "app";

    private String targetBundleName = "";

    private int targetPriority = 0;

    private String targetModuleName = "";

    private int targetModulePriority = 0;

    private List<PreloadItem> preloadItems = new ArrayList<>();

    private long fileLength = 0L;
    private int entrySizeLimit = 2048;

    private int notEntrySizeLimit = 2048;

    private int sumSizeLimit = 10240;

    private boolean debug = false;

    private String compileSdkVersion = "";

    private String compileSdkType = "";

    private List<String> proxyDataUris = new ArrayList<>();

    private Map<String, List<String>> continueTypeMap = new HashMap<>();

    private MultiAppMode multiAppMode = new MultiAppMode();

    private String fileType = "";

    /**
     * get bundle name form HapVerifyInfo.
     */
    public String getBundleName() {
        return bundleName;
    }

    /**
     * set bundle name for HapVerifyInfo.
     */
    public void setBundleName(String bundleName) {
        this.bundleName = bundleName;
    }

    /**
     * get vendor form HapVerifyInfo.
     */
    public String getVendor() {
        return vendor;
    }

    /**
     * set vendor for HapVerifyInfo.
     */
    public void setVendor(String vendor) {
        this.vendor = vendor;
    }

    /**
     * get version from HapVerifyInfo.
     */
    public Version getVersion() {
        return version;
    }

    /**
     * set version for HapVerifyInfo.
     */
    public void setVersion(Version version) {
        this.version = version;
    }

    /**
     * get apiVersion from HapVerifyInfo.
     */
    public ModuleApiVersion getApiVersion() {
        return apiVersion;
    }

    /**
     * set apiVersion for HapVerifyInfo.
     */
    public void setApiVersion(ModuleApiVersion apiVersion) {
        this.apiVersion = apiVersion;
    }

    /**
     * Get assetAccessGroups from HapVerifyInfo.
     */
    public List<String> getAssetAccessGroups() {
        return assetAccessGroups;
    }

    /**
     * Set assetAccessGroups for HapVerifyInfo.
     */
    public void setAssetAccessGroups(List<String> assetAccessGroups) {
        this.assetAccessGroups = assetAccessGroups;
    }

    /**
     * get module name from HapVerifyInfo.
     */
    public String getModuleName() {
        return moduleName;
    }

    /**
     * set module name for HapVerifyInfo.
     */
    public void setModuleName(String moduleName) {
        this.moduleName = moduleName;
    }

    /**
     * get package name from HapVerifyInfo.
     */
    public String getPackageName() {
        return packageName;
    }

    /**
     * set package name for HapVerifyInfo.
     */
    public void setPackageName(String packageName) {
        this.packageName = packageName;
    }

    /**
     * get ability names from HapVerifyInfo.
     */
    public List<String> getAbilityNames() {
        return abilityNames;
    }

    /**
     * set abilityNames for HapVerifyInfo.
     */
    public void setAbilityNames(List<String> abilityNames) {
        this.abilityNames = abilityNames;
    }

    /**
     * add abilityNames for HapVerifyInfo.
     */
    public void addAbilityNames(List<String> nameList) {
        this.abilityNames.addAll(nameList);
    }

    /**
     * get distroFilter from HapVerifyInfo.
     */
    public DistroFilter getDistroFilter() {
        return distroFilter;
    }

    /**
     * set distroFilter for HapVerifyInfo.
     */
    public void setDistroFilter(DistroFilter distroFilter) {
        this.distroFilter = distroFilter;
    }

    /**
     * get deviceType from HapVerifyInfo.
     */
    public List<String> getDeviceType() {
        return deviceType;
    }

    /**
     * set deviceType for HapVerifyInfo.
     */
    public void setDeviceType(List<String> deviceType) {
        this.deviceType = deviceType;
    }

    /**
     * get isStageModule from HapVerifyInfo.
     */
    public boolean isStageModule() {
        return isStageModule;
    }

    /**
     * set isStageModule for HapVerifyInfo.
     */
    public void setStageModule(boolean isStageModule) {
        this.isStageModule = isStageModule;
    }

    /**
     * get moduleType from HapVerifyInfo.
     */
    public String getModuleType() {
        return moduleType;
    }

    /**
     * set is module for HapVerifyInfo.
     */
    public void setModuleType(String moduleType) {
        this.moduleType = moduleType;
    }

    /**
     * get isInstallationFree form HapVerifyInfo.
     */
    public boolean isInstallationFree() {
        return isInstallationFree;
    }

    /**
     * set isInstallationFree for HapVerifyInfo.
     */
    public void setInstallationFree(boolean isInstallationFree) {
        this.isInstallationFree = isInstallationFree;
    }

    /**
     * get dependency form HapVerifyInfo.
     */
    public List<String> getDependencies() {
        return dependencies;
    }

    /**
     * set dependency for HapVerifyInfo.
     */
    public void setDependencies(List<String> dependencies) {
        this.dependencies = dependencies;
    }

    /**
     * get dependency item list for HapVerifyInfo.
     */
    public List<DependencyItem> getDependencyItemList() {
        return dependencyItemList;
    }

    /**
     * set dependency item list for HapVerifyInfo.
     */
    public void setDependencyItemList(List<DependencyItem> dependencyItemList) {
        this.dependencyItemList = dependencyItemList;
        convertToDependency();
    }

    /**
     * get json file string form HapVerifyInfo.
     */
    public String getProfileStr() {
        return profileStr;
    }

    /**
     * set json file string for HapVerifyInfo.
     */
    public void setProfileStr(String profileStr) {
        this.profileStr = profileStr;
    }

    /**
     * get resource map form HapVerifyInfo.
     */
    public HashMap<String, String> getResourceMap() {
        return resourceMap;
    }

    /**
     * set resource map for HapVerifyInfo.
     */
    public void setResourceMap(HashMap<String, String> resourceMap) {
        this.resourceMap = resourceMap;
    }

    private void convertToDependency() {
        for (DependencyItem item : dependencyItemList) {
            if (item.getBundleName() != null && item.getBundleName().equals(bundleName)) {
                dependencies.add(item.getModuleName());
            }
        }
    }

    public String getBundleType() {
        return bundleType;
    }

    public void setBundleType(String bundleType) {
        this.bundleType = bundleType;
    }

    public String getAtomicServiceType() {
        return atomicServiceType;
    }

    public void setAtomicServiceType(String atomicServiceType) {
        this.atomicServiceType = atomicServiceType;
    }

    public List<PreloadItem> getPreloadItems() {
        return preloadItems;
    }

    public void setPreloadItems(List<PreloadItem> preloadItems) {
        this.preloadItems = preloadItems;
    }

    public String getTargetBundleName() {
        return targetBundleName;
    }

    public void setTargetBundleName(String targetBundleName) {
        this.targetBundleName = targetBundleName;
    }

    public int getTargetPriority() {
        return targetPriority;
    }

    public void setTargetPriority(int priority) {
        this.targetPriority = priority;
    }

    public String getTargetModuleName() {
        return targetModuleName;
    }

    public void setTargetModuleName(String targetModuleName) {
        this.targetModuleName = targetModuleName;
    }

    public int getTargetModulePriority() {
        return targetModulePriority;
    }

    public void setTargetModulePriority(int priority) {
        this.targetModulePriority = priority;
    }

    public long getFileLength() {
        return fileLength;
    }

    public void setFileLength(long fileLength) {
        this.fileLength = fileLength;
    }

    public void setEntrySizeLimit(int limit) {
        this.entrySizeLimit = limit;
    }

    public int getEntrySizeLimit() {
        return entrySizeLimit;
    }

    public void setNotEntrySizeLimit(int notEntrySizeLimit) {
        this.notEntrySizeLimit = notEntrySizeLimit;
    }

    public int getNotEntrySizeLimit() {
        return notEntrySizeLimit;
    }

    public void setSumSizeLimit(int sumSizeLimit) {
        this.sumSizeLimit = sumSizeLimit;
    }

    public int getSumSizeLimit() {
        return sumSizeLimit;
    }

    public boolean isDebug() {
        return debug;
    }

    public void setDebug(boolean debug) {
        this.debug = debug;
    }

    public String getCompileSdkVersion() {
        return compileSdkVersion;
    }

    public void setCompileSdkVersion(String compileSdkVersion) {
        this.compileSdkVersion = compileSdkVersion;
    }

    public String getCompileSdkType() {
        return compileSdkType;
    }

    public void setCompileSdkType(String compileSdkType) {
        this.compileSdkType = compileSdkType;
    }

    public List<String> getProxyDataUris() {
        return proxyDataUris;
    }

    public void setProxyDataUris(List<String> proxyDataUris) {
        this.proxyDataUris = proxyDataUris;
    }

    /**
     * get continueType map for HapVerifyInfo.
     */
    public Map<String, List<String>> getContinueTypeMap() {
        return continueTypeMap;
    }

    /**
     * set continueType map for HapVerifyInfo.
     */
    public void setContinueTypeMap(Map<String, List<String>> continueTypeMap) {
        this.continueTypeMap = continueTypeMap;
    }

    public MultiAppMode getMultiAppMode() {
        return multiAppMode;
    }

    public void setMultiAppMode(MultiAppMode multiAppMode) {
        this.multiAppMode = multiAppMode;
    }

    public String getFileType() {
        return fileType;
    }

    public void setFileType(String fileType) {
        this.fileType = fileType;
    }
}

File: packing_tool/frameworks/src/json/hap_verify_info.cpp
/*
 * Copyright (c) 2024 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#include "hap_verify_info.h"

#include <iostream>

namespace OHOS {
namespace AppPackingTool {
const std::string& HapVerifyInfo::GetBundleName() const
{
    return bundleName_;
}

void HapVerifyInfo::SetBundleName(const std::string& bundleName)
{
    bundleName_ = bundleName;
}

const std::string& HapVerifyInfo::GetVendor() const
{
    return vendor_;
}

void HapVerifyInfo::SetVendor(const std::string& vendor)
{
    vendor_ = vendor;
}

const Version& HapVerifyInfo::GetVersion() const
{
    return version_;
}

void HapVerifyInfo::SetVersion(const Version& version)
{
    version_ = version;
}

const ModuleApiVersion& HapVerifyInfo::GetApiVersion() const
{
    return apiVersion_;
}

void HapVerifyInfo::SetApiVersion(const ModuleApiVersion& apiVersion)
{
    apiVersion_ = apiVersion;
}

const std::string& HapVerifyInfo::GetModuleName() const
{
    return moduleName_;
}

void HapVerifyInfo::SetModuleName(const std::string& moduleName)
{
    moduleName_ = moduleName;
}

const std::string& HapVerifyInfo::GetPackageName() const
{
    return packageName_;
}

void HapVerifyInfo::SetPackageName(const std::string& packageName)
{
    packageName_ = packageName;
}

const std::list<std::string>& HapVerifyInfo::GetAbilityNames() const
{
    return abilityNames_;
}

void HapVerifyInfo::SetAbilityNames(const std::list<std::string>& abilityNames)
{
    abilityNames_ = abilityNames;
}

void HapVerifyInfo::AddAbilityNames(const std::list<std::string>& nameList)
{
    abilityNames_.insert(abilityNames_.end(), nameList.begin(), nameList.end());
}

const DistroFilter& HapVerifyInfo::GetDistroFilter() const
{
    return distroFilter_;
}

void HapVerifyInfo::SetDistroFilter(const DistroFilter& distroFilter)
{
    distroFilter_ = distroFilter;
}

const std::list<std::string>& HapVerifyInfo::GetDeviceTypes() const
{
    return deviceTypes_;
}

void HapVerifyInfo::SetDeviceTypes(const std::list<std::string>& deviceTypes)
{
    deviceTypes_ = deviceTypes;
}

const bool& HapVerifyInfo::IsStageModule() const
{
    return isStageModule_;
}

void HapVerifyInfo::SetStageModule(const bool& isStageModule)
{
    isStageModule_ = isStageModule;
}

const std::string& HapVerifyInfo::GetModuleType() const
{
    return moduleType_;
}

void HapVerifyInfo::SetModuleType(const std::string& moduleType)
{
    moduleType_ = moduleType;
}

const bool& HapVerifyInfo::IsInstallationFree() const
{
    return isInstallationFree_;
}

void HapVerifyInfo::SetInstallationFree(const bool& isInstallationFree)
{
    isInstallationFree_ = isInstallationFree;
}

const std::list<std::string>& HapVerifyInfo::GetDependencies() const
{
    return dependencies_;
}

void HapVerifyInfo::SetDependencies(const std::list<std::string>& dependencies)
{
    dependencies_ = dependencies;
}

const std::list<DependencyItem>& HapVerifyInfo::GetDependencyItemList() const
{
    return dependencyItemList_;
}

void HapVerifyInfo::SetDependencyItemList(const std::list<DependencyItem>& dependencyItemList)
{
    dependencyItemList_ = dependencyItemList;
    ConvertToDependency();
}

const std::string& HapVerifyInfo::GetProfileStr() const
{
    return profileStr_;
}

void HapVerifyInfo::SetProfileStr(const std::string& profileStr)
{
    profileStr_ = profileStr;
}

const std::map<std::string, std::string>& HapVerifyInfo::GetResourceMap() const
{
    return resourceMap_;
}

void HapVerifyInfo::SetResourceMap(const std::map<std::string, std::string>& resourceMap)
{
    resourceMap_ = resourceMap;
}

const std::string& HapVerifyInfo::GetBundleType() const
{
    return bundleType_;
}

void HapVerifyInfo::SetBundleType(const std::string& bundleType)
{
    bundleType_ = bundleType;
}

const std::string& HapVerifyInfo::GetAtomicServiceType() const
{
    return atomicServiceType_;
}

void HapVerifyInfo::SetAtomicServiceType(const std::string& atomicServiceType)
{
    atomicServiceType_ = atomicServiceType;
}

const std::list<PreloadItem>& HapVerifyInfo::GetPreloadItems() const
{
    return preloadItems_;
}

void HapVerifyInfo::SetPreloadItems(const std::list<PreloadItem>& preloadItems)
{
    preloadItems_ = preloadItems;
}

const std::string& HapVerifyInfo::GetTargetBundleName() const
{
    return targetBundleName_;
}

void HapVerifyInfo::SetTargetBundleName(const std::string& targetBundleName)
{
    targetBundleName_ = targetBundleName;
}

const int32_t& HapVerifyInfo::GetTargetPriority() const
{
    return targetPriority_;
}

void HapVerifyInfo::SetTargetPriority(const int32_t& priority)
{
    targetPriority_ = priority;
}

const std::string& HapVerifyInfo::GetTargetModuleName() const
{
    return targetModuleName_;
}

void HapVerifyInfo::SetTargetModuleName(const std::string& targetModuleName)
{
    targetModuleName_ = targetModuleName;
}

const int32_t& HapVerifyInfo::GetTargetModulePriority() const
{
    return targetModulePriority_;
}

void HapVerifyInfo::SetTargetModulePriority(const int32_t& priority)
{
    targetModulePriority_ = priority;
}

const int64_t& HapVerifyInfo::GetFileLength() const
{
    return fileLength_;
}

void HapVerifyInfo::SetFileLength(const int64_t& fileLength)
{
    fileLength_ = fileLength;
}

const int32_t& HapVerifyInfo::GetEntrySizeLimit() const
{
    return entrySizeLimit_;
}

void HapVerifyInfo::SetEntrySizeLimit(const int32_t& limit)
{
    entrySizeLimit_ = limit;
}

const int32_t& HapVerifyInfo::GetNotEntrySizeLimit() const
{
    return notEntrySizeLimit_;
}

void HapVerifyInfo::SetNotEntrySizeLimit(const int32_t& notEntrySizeLimit)
{
    notEntrySizeLimit_ = notEntrySizeLimit;
}

const int32_t& HapVerifyInfo::GetSumSizeLimit() const
{
    return sumSizeLimit_;
}

void HapVerifyInfo::SetSumSizeLimit(const int32_t& sumSizeLimit)
{
    sumSizeLimit_ = sumSizeLimit;
}

const bool& HapVerifyInfo::IsDebug() const
{
    return debug_;
}

void HapVerifyInfo::SetDebug(const bool& debug)
{
    debug_ = debug;
}

const std::string& HapVerifyInfo::GetCompileSdkVersion() const
{
    return compileSdkVersion_;
}

void HapVerifyInfo::SetCompileSdkVersion(const std::string& compileSdkVersion)
{
    compileSdkVersion_ = compileSdkVersion;
}

const std::string& HapVerifyInfo::GetCompileSdkType() const
{
    return compileSdkType_;
}

void HapVerifyInfo::SetCompileSdkType(const std::string& compileSdkType)
{
    compileSdkType_ = compileSdkType;
}

const std::list<std::string>& HapVerifyInfo::GetProxyDataUris() const
{
    return proxyDataUris_;
}

void HapVerifyInfo::SetProxyDataUris(const std::list<std::string>& proxyDataUris)
{
    proxyDataUris_ = proxyDataUris;
}

const std::map<std::string, std::list<std::string>>& HapVerifyInfo::GetContinueTypeMap() const
{
    return continueTypeMap_;
}

void HapVerifyInfo::SetContinueTypeMap(const std::map<std::string, std::list<std::string>>& continueTypeMap)
{
    continueTypeMap_ = continueTypeMap;
}

const MultiAppMode& HapVerifyInfo::GetMultiAppMode() const
{
    return multiAppMode_;
}

void HapVerifyInfo::SetMultiAppMode(const MultiAppMode& multiAppMode)
{
    multiAppMode_ = multiAppMode;
}

const std::list<std::string>& HapVerifyInfo::GetAssetAccessGroups() const
{
    return assetAccessGroups_;
}

void HapVerifyInfo::SetAssetAccessGroups(const std::list<std::string>& assetAccessGroups)
{
    assetAccessGroups_ = assetAccessGroups;
}

void HapVerifyInfo::ConvertToDependency()
{
    for (DependencyItem item : dependencyItemList_) {
        if (!item.bundleName.empty() && item.bundleName.compare(bundleName_) == 0) {
            dependencies_.emplace_back(item.bundleName);
        }
    }
}

const std::string& HapVerifyInfo::GetFileType() const
{
    return fileType_;
}

void HapVerifyInfo::SetFileType(const std::string& fileType)
{
    fileType_ = fileType;
}
} // namespace AppPackingTool
} // namespace OHOS
